<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>540</width>
    <height>540</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>540</width>
    <height>540</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>540</width>
    <height>540</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>RF test</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>540</width>
     <height>540</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>540</width>
     <height>540</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item alignment="Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTop">
       <widget class="QWidget" name="programmerWidget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item alignment="Qt::AlignmentFlag::AlignLeft">
          <widget class="QWidget" name="RFTestVersionWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="rfTestVersionLabel">
              <property name="text">
               <string>RF Test version:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="rfTestVersion">
              <property name="text">
               <string>TextLabel</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item alignment="Qt::AlignmentFlag::AlignLeft">
          <widget class="QWidget" name="donleFWVersionWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="dongleFWVersionLabel">
              <property name="text">
               <string>Dongle FW version:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="dongleFWVersion">
              <property name="text">
               <string>TextLabel</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QFormLayout" name="programForm">
           <item row="0" column="0">
            <widget class="QLabel" name="debuggerLabel">
             <property name="text">
              <string>Debugger</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="debuggerSNR"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="deviceVersionLabel">
             <property name="text">
              <string>Device version:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="deviceVersion"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="loadCapacitorLabel">
             <property name="text">
              <string>HFXO Load capacitor:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="loadCapacitor">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Internal load capacitors in pF. To disable the internal load capacitors set this to 0. Leave blank to use default configuration.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QWidget" name="radioButtons" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QRadioButton" name="FEM">
              <property name="text">
               <string>FEM</string>
              </property>
              <property name="autoExclusive">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="autoTest">
              <property name="text">
               <string>Automatic</string>
              </property>
              <property name="autoExclusive">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="femPinWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item alignment="Qt::AlignmentFlag::AlignHCenter">
             <widget class="QLabel" name="femPinsLabel">
              <property name="text">
               <string>FEM pin config</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QFormLayout" name="femPins">
              <item row="0" column="0">
               <widget class="QLabel" name="txEnLabel">
                <property name="text">
                 <string>TX_EN</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="txEnPin">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;TX enable pin. High when DUT transmitting, low otherwise.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="rxEnLabel">
                <property name="text">
                 <string>RX_EN</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="rxEnPin">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;RX enable pin. High when DUT receiving, low otherwise.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="pdnLabel">
                <property name="text">
                 <string>PDN</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="pdnPin">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Power down pin, always high.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="modeLabel">
                <property name="text">
                 <string>MODE</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLineEdit" name="modePin">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Mode pin, state controlled by FEM config and can be changed without reflashing test firmware.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="antSelLabel">
                <property name="text">
                 <string>ANT_SEL</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QLineEdit" name="antSelPin">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Antenna select pin, state controlled by FEM config and can be changed without reflashing test firmware.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="buttonWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <layout class="QVBoxLayout" name="buttonRows">
              <item>
               <layout class="QHBoxLayout" name="buttonRow1">
                <item>
                 <widget class="QPushButton" name="flashButton">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flash test firmware to DUT, and writes load capacitor config and FEM pin config to DUT.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string>Flash</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="recoverButton">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Recover the DUT by erasing flash and disabling readback protection.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string>Recover</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="resetButton">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Pin reset, requires DUT reset pin to be connected to the debugger.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string>Reset</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="buttonRow2">
                <item>
                 <widget class="QPushButton" name="detectDeviceButton">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Gets device version of the connected DUT.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string>Detect device</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="updateDebuggersButton">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Gets all debuggers available.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="text">
                   <string>Get debuggers</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_3">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop">
       <widget class="QWidget" name="rfTestWiget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QWidget" name="testConfigWidget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item alignment="Qt::AlignmentFlag::AlignHCenter">
             <widget class="QLabel" name="testConfigLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Test config</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QFormLayout" name="testConfig">
              <item row="0" column="0">
               <widget class="QLabel" name="testLabel">
                <property name="text">
                 <string>Test</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="testType">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="dataRateLabel">
                <property name="text">
                 <string>Data Rate</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QComboBox" name="dataRate"/>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="outputPowerLabel">
                <property name="text">
                 <string>Output Power</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="outputPower">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Output power in dBm. If the configured output power isn't possible by the radio, the closest possible radio output power not exeeding the configured output power will be used.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="firstChannelLabel">
                <property name="text">
                 <string>Frequency</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLineEdit" name="firstChannel">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Radio frequency in MHz offset from 2400MHz.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="lastChannelLabel">
                <property name="text">
                 <string>Stop Frequency</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QLineEdit" name="lastChannel">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Ending sweep frequency in MHz offset from 2400MHz.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="femSettingsWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item alignment="Qt::AlignmentFlag::AlignHCenter">
             <widget class="QLabel" name="pinSettingsLabel">
              <property name="text">
               <string>FEM config</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QFormLayout" name="femSettings">
              <item row="0" column="0">
               <widget class="QLabel" name="modeSettingLabel">
                <property name="text">
                 <string>Mode</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="modeSetting">
                <property name="wrapping">
                 <bool>true</bool>
                </property>
                <property name="maximum">
                 <number>1</number>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="antSelSetting">
                <property name="wrapping">
                 <bool>true</bool>
                </property>
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>0</number>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="antSelSettingLabel">
                <property name="text">
                 <string>ANT_SEL</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="autoTestWidget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item alignment="Qt::AlignmentFlag::AlignHCenter">
             <widget class="QLabel" name="autoConfigLabel">
              <property name="text">
               <string>Auto config</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QFormLayout" name="autoConfig">
              <item row="1" column="0">
               <widget class="QLabel" name="voltageLabel">
                <property name="text">
                 <string>Voltage</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="voltageLineEdit"/>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="channelsLabel">
                <property name="text">
                 <string>Channels</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="channelsLineEdit"/>
              </item>
              <item row="3" column="1">
               <widget class="QLineEdit" name="powersLineEdit"/>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="powersLabel">
                <property name="text">
                 <string>Powers</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="startButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start RF test.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="toolTipDuration">
            <number>-1</number>
           </property>
           <property name="text">
            <string>Start</string>
           </property>
           <property name="shortcut">
            <string>Return</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>540</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuConfiguration">
    <property name="title">
     <string>Configuration</string>
    </property>
    <addaction name="actionSpectrum_analyzer"/>
    <addaction name="actionPower_supply"/>
   </widget>
   <addaction name="menuConfiguration"/>
  </widget>
  <action name="actionSpectrum_analyzer">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Spectrum analyzer</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="actionPower_supply">
   <property name="text">
    <string>Power supply</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>startButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
